src/shared/lib/GenericAssayUtils/MutationalSignaturesUtils.tsx

```diff
@@ -163,3 +166,42 @@ export function validateMutationalSignatureRawData(
     // we are expecting contribution and pvalue profiles are in pairs
     return _.every(profileIdsGroupByVersion, ids => ids.length === 2);
 }
+
+export function retrieveMutationalSignatureVersionFromData(
+    signatureProfiles: string[]
+): string {
+    const uniqueProfileVersion = _.uniq(
+        signatureProfiles.map(function(obj) {
+            return _.last(obj.split('_'));
+        })
+    );
+    if (uniqueProfileVersion !== undefined) {
+        // give priority to version 3 over version 2
+        if (
+            uniqueProfileVersion.includes('v3') &&
+            uniqueProfileVersion.includes('v2')
+        ) {
+            const valueToSet: string = 'v3';
+            return valueToSet;
+        } else {
+            // Can we make this easier? (so just select the version present
+            // To also take into account when there are version outside the enumType)
+            const versionToSet = Object.values(
```

Comment:
```versionToSet = Object.values(MutationalSignaturesVersion).find(uniqueProfileVersion[0])
```
then you can test for undefined.

Reply:
in fact i think this whole thing can be reduced to:

```
return Object.values(MutationalSignaturesVersion)
.find(uniqueProfileVersion[0]) ? uniqueProfileVersion[0]
```

Reply:
I've reduced the code and removed the if/else statements. I believe it can be reduced even more if we don't force the version to be a member of the enum MutationalSignaturesVersion. This could be beneficial when a version exists that is not yet part of MutationalSignaturesVersion. Or, when users use _early or _late to distinguish between mutational signature profiles (and not _v2 or _v3)

The code becomes:
`return (uniqueProfileVersion.includes('v3') && uniqueProfileVersion.includes('v2')) ? 'v3' : uniqueProfileVersion[0]!`

Reply:
@TJMKuijpers @pvannierop can you comment on the above?

Reply:
In the original code, the _selectedMutationalSignatureVersion is set to version 2 by:
`@observable _selectedMutationalSignatureVersion: string =MutationalSignaturesVersion.V2;`
When loading the mutational signature tab, the version is retrieved via:
```
       version={
                pageComponent.patientViewPageStore.selectedMutationalSignatureVersion
}
```
which will call :
```
		@computed get selectedMutationalSignatureVersion() {
        				return this._selectedMutationalSignatureVersion;
    			}
```
By default this will be v2, which causes a problem when version 2 data is not present (page will turn blank). 

To fix the dependency on version 2, we could check if the mutational signature version of a study belongs to enum MutationalSignatureVersion:

```
export enum MutationalSignaturesVersion {
    V2 = 'v2',
    V3 = 'v3',
}
```
However, this could be a problem if the molecularProfileId of our mutational signature data does not end with _v2 and v3 (because of the string split in availableVersions()). For example, we could have 3 mutational signature files with a molecularProfileId _SBS, _DBS, and _ID. To solve this problem, we could add SBS, DBS, and ID to MutationalSignaturesVersion. However, this means that when we have a molecularProfileId, not in MutationalSignaturesVersion, we have to update it to be able to load the mutational signature data.

By having a function that is independent of MutationalSignaturesVersion, we don't have to worry about molecularProfileIds that are not a member (or in this case, are v2 or v3).

`return (uniqueProfileVersion.includes('v3') && uniqueProfileVersion.includes('v2')) ? 'v3' : uniqueProfileVersion[0]!`
In this case, we give priority over version 3 if both versions 2 and 3 are present. Otherwise, we will return the first value of uniqueProfileVersion. This could be either v2, v3, or any other version (if we regard SBS, DBS, and ID COSMIC signatures as a version).